### ISSUE TITLE
documentation is oversimplified and not really useful

### ISSUE THREAD

**Issue by matanox** [mostlyserious/riak-js#206]
The documentation site over-simplifies things, e.g. does not mention that `save` is asynchronous. Looking at the repo tests is more useful than reading the api documentation. To consider removing the documentation site or rewriting it for 2.0.

**Comment by rm-rf-etc** [comment]
I agree with the above points. I'm currently conflicted about which riak client–riak-js or nodiak–I am better off using and supporting. While this project has a good amount of activity in the issues section and a long list of people who have made small contributions, the docs and lower readability of the code itself, along with nodiak being listed / linked on the basho website (and riak-js not), has given me reason to consider nodiak more so than riak-js. I've found it easy to add support to nodiak for fetching objects using riak 2.0 bucket-types.

The main issue that I take with riak-js.org is that the examples don't cover the basics first. Document creation needs to come before metas, protobuffs, load balancing, etc. As a first-time NoSQL user, I want to create a user object, and then create a few more associated objects, do some kind of relational lookup (search), and make some updates. Instead, the order on riak-js.org is:
• create a client
• do something with a protocol buffer (I don't even know what these are yet)
• load balancing and connection pooling (I don't have a load to balance yet)
• fetch a document
• meta objects
• ...
It seems there isn't a section on creating documents. The first mention of a PUT appears to be in the section "Callbacks".

The history log in the readme is nice but would be better as a separate doc, with a 'hello world' example in the readme.

**Comment by janearc** [comment]
Please note that "being listed on the basho website" only requires submitting a PR and doesn't imply any endorsement or suitability.
